### PR TITLE
[ui] Add system setting option for themes

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Allow using the system setting for light/dark mode. This is added as a "System setting" option in the theme selector.

Also add a product tour component to direct users to the feature. Track its dismissed state in localStorage.

Screenshot 2023-12-11 at 4 39 22 PM

## How I Tested These Changes

Remove `dagster-theme` from localStorage, view app. Verify that the tour popover appears, and that it can be dismissed (with state tracked) by clicking "Dismiss".

Choose "System setting" from theme selector in  User Settings dialog, click "Done". Verify that I see dark mode on reload, with "System" as the value in localStorage.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18655** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18655?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18655?utm_source=stack-comment).

**Comment by salazarm** [comment]
The little triangle in the top right is white

**Comment by hellendag** [comment]
It sure is, good catch! Probably a mixup in the change to themes. Will fix.

**Comment by braunjj** [comment]
Love it. Is there anyway to force a refresh? It's slightly annoying that I have to manually do a browser refresh for the theme change to take effect.

File: js_modules/dagster-ui/packages/ui-core/src/app/UserSettingsButton.tsx [hunk 1]
```diff
@@ -34,11 +44,51 @@ const SettingsButton = styled.button`
...
+      }
+      actions={{
+        dismiss: () => {
+          setShouldHideUIThemeNux('1');
+        },
+      }}
+      modifiers={{offset: {enabled: true, options: {offset: [-12, 8]}}}}
+    >
+      {button}
+    </ProductTour>
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/app/UserSettingsButton.tsx:86]: Wow I think I'd missed this in other PRs, this is a really nice component interface 💯

File: js_modules/dagster-ui/packages/ui-components/src/theme/theme.tsx [hunk 2]
```diff
@@ -28,15 +29,22 @@ export const getTheme = memoize(() => {
...
+  ) {
     return value;
   }
 
-  // todo dish: Uncomment this to allow OS settings to work automatically.
-  // if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
-  //   return DagsterTheme.Dark;
-  // }
+  if (value === DagsterTheme.System) {
+    if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-components/src/theme/theme.tsx:41]: This looks good to me! I like that the memoize means this is only re-computed on page load so systems that change at dusk won't start returning `dark` as individual components on the page re-render.

**Review by bengotow — APPROVED** [review]
Looks great! Will switch to using this after it merges :-)

**Comment by hellendag** [comment]
Fixed the tour arrow.

Screenshot 2023-12-12 at 9 47 25 AM
Screenshot 2023-12-12 at 9 47 07 AM

**Comment by hellendag** [comment]
> Love it. Is there anyway to force a refresh? It's slightly annoying that I have to manually do a browser refresh for the theme change to take effect.

Yeah, I have something in mind for this.

**Pull request merged**